client/me/purchases/manage-purchase/index.jsx

```diff
@@ -312,7 +312,7 @@ class ManagePurchase extends Component {
 			description = theme.description;
 		} else if ( isDomainMapping( purchase ) || isDomainRegistration( purchase ) ) {
 			description = translate(
-				"Replaces your site's free address with the domain %(domain)s, " +
+				"Replaces your site's free address %(domain)s with the domain, " +
```

Comment:
Since the `%(domain)s` is a non-defining detail, it should be enclosed in comma's.  E.g,

"Replaces your site's free address, example.wordpress.com, with the domain, making it easier to remember and share"